azext_edge/tests/edge/checks/int/helpers.py

```diff
@@ -227,3 +254,12 @@ def run_check_command(
     result = run(command)
 
     return {cond["name"]: cond for cond in result["postDeployment"]}, service_present
+
+
+def condition_statuses(
+    conditions: list,
+    known_conditions: list,
+):
+    for condition in conditions:
+        if condition['type'] in known_conditions:
+            yield condition['status'] == 'True'
```

Comment:
What is the reason for use of `yield` here ?

Reply:
updated to use return